### DISCUSSION TITLE
Grid - Edit / Delete / Single Selection / Multi Selection

### DISCUSSION THREAD

**Discussion by devrushnet** [vikramlearning/blazorbootstrap#176] [Q&A]
Hi,

I'm testing your grid component, how to insert the follow elements ?

1) Colum button (for edit/delete) on the Row
2) Single Selection Row
3) Multi Selection Rows

best regards

**Comment by gvreddy04** [comment]
@devrushnet Thanks for testing our components. I'll share a sample code ASAP.

**Reply by devrushnet** [reply]
Hi, 
the solution for the check it's no good, the grid must have a check button in the col header for the select all items.
It's possibile insert the properties SelectionMode = for Single and MultiSolution (any grid for single solution have select row without check button)?

best regards

**Reply by gvreddy04** [reply]
@devrushnet This is an interim solution for now. We will add Edit/Delete/Single selection/Multi Selection features ASAP.

**Comment by gvreddy04** [comment]
@devrushnet 

Sample GitHub Project: https://github.com/vikramlearning/blazorbootstrap-examples

Custom Grid Operations:
- Edit
- Delete
- Single Row Selection
- Multiple Rows Selection

## Employees Grid
![image](https://user-images.githubusercontent.com/2337067/230451580-18ab4544-15f7-4e1c-b991-9b4424a1a0a2.png)

## Edit Employee
![image](https://user-images.githubusercontent.com/2337067/230451895-dad8cbeb-63b2-4834-8b7b-dfc3c71d2e81.png)

## Update Employee
![image](https://user-images.githubusercontent.com/2337067/230452078-a795a940-690b-48dc-ba58-72e35550d0d4.png)

## Delete Employee
![image](https://user-images.githubusercontent.com/2337067/230452303-41bf0c43-ce32-4f93-be7c-247775a5c930.png)

## After Delete
![image](https://user-images.githubusercontent.com/2337067/230452371-acf6d8ef-0967-4819-9b2b-ddc5722033be.png)

## Single / Multiple Rows Selection
We can write custom logic to check/uncheck the checkbox. Based on the checkbox selection, you can change the row color.

## References:
- Conditional css class for grid row: https://demos.blazorbootstrap.com/grid#conditional-css-class-for-grid-row
- Conditional css class for grid column: https://demos.blazorbootstrap.com/grid#conditional-css-class-for-grid-column

I hope this helps.

**Comment by lukaskonvicka** [comment]
Hello,

Im looking for an example with edit/delete button, but I can not find it in this example project https://github.com/vikramlearning/blazorbootstrap-examples.

Am I looking at wrong place?

Thank you
Lukas

**Reply by lukaskonvicka** [reply]
Hello, 
I just found it, sorry for bothering.
Lukas
